### ISSUE TITLE
X310. When turning on TX and RX in one prog, RX sleeping for a minute. To check, run txrx_loopback_to_file  --file usrp_samples.dat --tx-rate 1000000 --rx-rate 1000000 --tx-freq 950000000 --rx-freq 950000000  --tx-ant="TX/RX" --rx-ant="TX/RX" --ref="internal"

### ISSUE THREAD

**Issue by medvedvvs** [EttusResearch/uhd#351]

**Comment by medvedvvs** [comment]
X310, PCIe. When turning on TX and RX in one prog, RX is sleeping for a minute. To check, run txrx_loopback_to_file  --file usrp_samples.dat --tx-rate 1000000 --rx-rate 1000000 --tx-freq 950000000 --rx-freq 950000000  --tx-ant="TX/RX" --rx-ant="TX/RX" --ref="internal"

**Comment by atrnati** [comment]
Would you be able to provide more details on your setup, including exact version of UHD you're using, system details (OS and version), and USRP hardware configuration (device, daughterboard(s), transports, etc.)?

**Comment by NI-LAm** [comment]
Can one do TX and RX on the same antenna port at the same time? Shouldn't the arguments be `--tx-ant="TX/RX --rx-ant="RX2"`? Configured like above I would assume the RX to wait for the TX to finish transmitting which might explain why there is a delay in RX.

**Comment by michaelld** [comment]
> Can one do TX and RX on the same antenna port at the same time?

Nope. Can do -either- TX -or- RX.

**Comment by michaelld** [comment]
@NI-LAm I stand corrected: I am told that one -can- do -simultaneous- TX and RX on the same TX/RX port (at least on the X310; I'm checking for others). The TX will bleed into the RX and so your app will need to find a way to ignore or filter the VERY LOUD TX'd signal when doing RX. To be honest: I just learned of this use option; I've never tried it myself; I'm a little surprised that this works! That said: I can see some pretty cool use cases for it, with the correct RX filtering out of the TX signal.

**Comment by michaelld** [comment]
See for example this [GRCon 2017 presentation](https://www.gnuradio.org/grcon/grcon17/presentations/full_duplex_radio/Adam-Parower-Full-Duplex-Radio.pdf) discussing the topic of how to do self-interference cancellation when using the same port for TX and RX.

**Comment by michael-west** [comment]
Yes, you can use the same port simultaneously for TX and RX.  The RF switch for the TX/RX port will be switched to the TX path and the signal on the RX path will be attenuated by the amount of the isolation provided by the switch.

@medvedvvs This could be an issue on the PCIe transport.  Are you able to test using 10 GbE?

**Comment by michael-west** [comment]
Redirecting to Support for reproduction and validation.
